app/cronjob/trailingTrade/step/place-buy-order.js

```diff
@@ -438,21 +432,38 @@ const execute = async (logger, rawData) => {
   // Set last buy grid order to be checked until it is executed
   await saveGridTradeOrder(logger, `${symbol}-grid-trade-last-buy-order`, {
     ...orderResult,
-    currentGridTradeIndex,
-    nextCheck: moment().add(checkOrderExecutePeriod, 'seconds').format()
+    currentGridTradeIndex
   });
 
   // Save number of open orders
   await saveOrderStats(logger, symbols);
 
   // Get open orders and update cache
-  data.openOrders = await getAndCacheOpenOrdersForSymbol(logger, symbol);
+  data.openOrders =
+    JSON.parse(await cache.hget('trailing-trade-open-orders', symbol)) || [];
```

Comment:
@habibalkhabbaz 

I was writing the test and realised, this line may be obsoleted or need to be reverted.
I want to ask your opinion. Please let me know if I misunderstood your intention or happy to change it.

The original code `getAndOpenOrdersForSymbol` was to get open orders immediately after placing/cancelling the orders. So it can be shown in the UI as well as processed with updated open orders in the next tick.

I can see you changed to get the open orders from the cache because the open orders will be cached whenever there is a new web socket received for `executionReport` - https://github.com/chrisleekr/binance-trading-bot/pull/431/files#diff-e0fe6e2b102cc13bc1813df41893cf958dc0bc1fa194f58f9bb101ace4be2b58R105

So there can be two solutions to optimise the code:

**Solution 1:**
- We don't need to set `data.openOrders` in the existing steps. We can add additional steps to set `data.openOrders`, `data.buy.openOrders`, and `data.sell.openOrders` with additional step at the end of the process. 
- The additional step will be getting open orders from the cache, not from API because we are expecting the web socket to already cached the open orders.
- My concern is, that there could have a timing gap when you get the open orders via WebSocket. If for some reason, WebSocket sends `executionReport` later, then the bot may not recognise there is an open order.

**Solution 2:**
- We revert back to `getAndCacheOpenOrdersForsymbol` to make sure open orders are up to date.
- The concern is, that API usage may increase. But I assume it will only increase when there is an action.

Please let me know what you think. Or if you have a better idea, please suggest me.

Reply:
Hello @chrisleekr 

I can't remember well for what reason I changed `getAndCacheOpenOrdersForSymbol` to `cache`.
The only main reason that I remember is to lower the API usage and to make the loop on symbols faster.
But I just realized now that I made the `executeTrailingTrade` independent! That's mean it will remain fast.
You are right and it's better to revert it back to `getAndCacheOpenOrdersForSymbol`.

Actually, this is the same thing for account balances but this is not affected because it can be tricked easily by subtraction or addition if you noticed  on `place-buy-order.js` file better than calling the API.

Last but not least, both solutions are good. However I prefer **Solution 2** because it make sense and I don't think it will increase the API usage too much since it will be called based on actions only.

Let me know if you would like me to do the changes. I will be happy to commit those changes.

Reply:
Good good. 

I will try to revert the code tonight. :) 
If you can before me, you can do it first.

Reply:
I will be happy to work on this but unfortunately the time here 3 AM now and maybe you will do it before me unless you became busy tonight then just keep it on me, no worries.

Reply:
Haha, go to bed. 
Let's see who can work on it first. :) 
Just notify before working it to avoid duplicate work.

Reply:
@habibalkhabbaz 
I will start reverting it.

Reply:
Done.